build.sbt

```diff
@@ -46,17 +48,28 @@ ThisBuild / githubWorkflowPublish := Seq(
   )
 )
 
+val nativeSettings = Def.settings(
+  scalaVersion := "3.1.0",
+  libraryDependencies += "org.scala-native" %%% "junit-runtime" % nativeVersion,
+  addCompilerPlugin("org.scala-native" % "junit-plugin" % nativeVersion cross CrossVersion.full),
+  testOptions += Tests.Argument(TestFrameworks.JUnit, "-a", "-s", "-v"),
+)
+
 // Aliases
 
-addCommandAlias("validate", ";clean;validateJVM;validateJS")
+addCommandAlias("validate", ";clean;validateJVM;validateJS;validateNative")
 addCommandAlias("validateJVM", ";buildJVM;mimaJVM;testJVM")
 addCommandAlias("validateJS", ";buildJS;mimaJS;testJS")
+addCommandAlias("validateNative", ";buildNative;mimaNative;testNative")
 addCommandAlias("buildJVM", ";derivingJVM/compile;testJVM/compile;typeableJVM/compile")
 addCommandAlias("buildJS", ";derivingJS/compile;testJS/compile;typeableJS/compile")
+addCommandAlias("buildNative", ";derivingNative/compile;testNative/compile")
```

Comment:
We're missing typeable - is that on purpose?

Reply:
Could not `typeableNative/Test/compile` 😢 

```
modules/typeable/src/test/scala/shapeless3/typeable/typeable.scala:315:32 
[error] 315 |    val co = o.cast[Option[Int]]
[error]     |                                ^
[error]     |Exception occurred while executing macro expansion.
[error]     |scala.scalanative.runtime.UndefinedBehaviorError
[error]     |	at scala.scalanative.runtime.package$.throwUndefined(package.scala:80)
[error]     |	at scala.scalanative.runtime.package$.intrinsic(package.scala:26)
[error]     |	at scala.scalanative.runtime.Intrinsics$.classFieldRawPtr(Intrinsics.scala:145)
[error]     |	at shapeless3.typeable.ReflectionUtils.Mirror(reflectionutils.scala:23)
[error]     |	at shapeless3.typeable.TypeableMacros$.mkSumTypeable$1(typeable.scala:337)
[error]     |	at shapeless3.typeable.TypeableMacros$.impl(typeable.scala:405)
[error]     |
[error]     | This location contains code that was inlined from typeable.scala:315
[error]     | This location contains code that was inlined from typeable.scala:250
```

Reply:
Does it mean macros don't work with Scala native? 🤔

Reply:
Didn't find anything in the release notes about macros: https://scala-native.readthedocs.io/en/latest/changelog/0.4.3-RC1.html - should we report it?

Reply:
Hey, it looks like a bug. `classFieldRawPtr` is currently only used in rewrites of LazyVals internal handling to allow for them to be run in Scala Native (code produced by Scala 3 for lazy vals is using sun.misc.Unsafe which would not be supported in Scala Native). I'll try to take a look at this issue and fix it before RC2

Reply:
In general Scala Native should work well with macros, they're evaluated long before SN is added to the compilation pipeline

Reply:
The fix for the compilation of typeable macros would be included in RC2, after applying the fix locally I was able to compile and successfully run all the tests.

Reply:
- https://github.com/typelevel/shapeless-3/pull/62 🎉